Automattic/jetpack
# Social: Add share status resharing

Fixes https://github.com/Automattic/jetpack-reach/issues/565

## Proposed changes:

* Added the retry action
* Filtered connections that are not available

### Other information:


## Jetpack product discussion


https://github.com/Automattic/jetpack-reach/issues/565
## Does this pull request change what data or activity we track or use?


No
## Testing instructions:






* Go to editor, share to all connections and without image to see a failed share for Instagram.
* You should see the retry action in the share log for that post.
* Click retry
* Wait for the endpoint to finish, then in the console run `await wp.data.dispatch( 'jetpack-social-plugin').invalidateResolution( 'getPostShareStatus', [<POST_ID>] );` to invalidate the response, and see the new share in the log
* Make sure it's only shared to the one that was retried

Disconnect an account from your blog that you previously shared to
* In the share log you should see a disconnected icon at the actions with a tooltip that it was removed. This only happens for failed ones

![CleanShot 2024-09-02 at 11 10 43 png](https://github.com/user-attachments/assets/6ca30044-16aa-4388-9f08-2bd3bf18272b)
